core/lib/zksync_core/src/fee_model.rs

```diff
@@ -64,11 +64,15 @@ impl BatchFeeModelInputProvider for MainNodeFeeInputProvider {
                 config,
                 l1_gas_price: self.provider.estimate_effective_gas_price(),
             }),
-            FeeModelConfig::V2(config) => FeeParams::V2(FeeParamsV2 {
-                config,
-                l1_gas_price: self.provider.estimate_effective_gas_price(),
-                l1_pubdata_price: self.provider.estimate_effective_pubdata_price(),
-            }),
+            FeeModelConfig::V2(config) => {
+                let validium_mode = std::env::var("VALIDIUM_MODE") == Ok("true".to_owned());
+                FeeParams::V2(FeeParamsV2 {
+                    config,
+                    l1_gas_price: self.provider.estimate_effective_gas_price(),
+                    l1_pubdata_price: self.provider.estimate_effective_pubdata_price()
+                        * !validium_mode as u64,
+                })
+            }
```

Comment:
This can be left untouched. The provider's `estimate_effective_pubdata_price` should return `0` instead in validium mode.

Reply:
IIRC that's a configuration flag in `chain.toml`.